src/App.js

```diff
@@ -1,19 +1,128 @@
 import React from 'react';
+import { TodoList } from './components/TodoList/TodoList';
 import './App.css';
 
 import users from './api/users';
+import todos from './api/todos';
 
-function App() {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
+const todosFromServer = todos.map(todo => ({
+  ...todo,
+  userName: users.find(user => user.id === todo.userId).name,
+}));
+
+class App extends React.Component {
+  state = {
+    todoList: todosFromServer,
+    task: '',
+    userName: '',
+    id: todosFromServer[todosFromServer.length - 1].id + 1,
+    errorUser: false,
+    errorTask: false,
+  };
+
+  handleChange = (event) => {
+    const { value, name } = event.target;
+
+    this.setState({
+      [name]: value.replace(/[^ a-z ]/gi, ''),
+      errorUser: false,
+      errorTask: false,
+    });
+  }
+
+  hendleSubmit = (event) => {
+    event.preventDefault();
+
+    const { task, userName } = this.state;
+
+    if (!task) {
+      this.setState({ errorTask: true });
+
+      return;
+    }
+
+    if (!userName || userName === 'Choose a user') {
+      this.setState({ errorUser: true });
+
+      return;
+    }
+
+    this.setState((state) => {
+      const newTodo = {
+        name: state.userName,
```

Comment:
```suggestion
        userName: state.userName,
```

user's name is undefined, cause you first save the value to 'name' prop, but read in Todo as 'userName'

![image](https://user-images.githubusercontent.com/57598825/106135546-be304d00-6170-11eb-9e36-69f238fd3610.png)